src/app/children/children-list/children-list.component.ts

```diff
@@ -88,6 +87,27 @@ export class ChildrenListComponent implements OnInit, AfterViewInit {
       .subscribe(results => this.prepareAttendanceData(results));
   }
 
+  getCsvExport() {
+    let csvData = '';
+    const rowNames = 'ProjectNumber,Name,Age,Date of Birth,Gender,Class,Center,Status,Mother Tongue,Religion,Address,Phone';
+    csvData += rowNames + '\r\n';
+    console.log(this.childrenList);
+    this.childrenList.map(child => {
+      csvData += child.projectNumber + ',' + child.name + ',' + child.age + ','
```

Comment:
Did you have a look at the PapaParse library: https://www.papaparse.com/docs#json-to-csv ?

Using their functions we could probably avoid listing all the fields of the Child object explicitly (and having to update this code every time the Child class gets another attribute).